src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -430,8 +437,10 @@ public void submit(Runnable r)
     //logger.debug("{}.{}.{}", currentThread, r, eventThread);
     if (eventThread == currentThread && tasks.isEmpty()) {
       r.run();
-    }
-    else {
+    } else {
+      if (!isActive()) {
```

Comment:
Do you want to allow addition of tasks when the event loop has stopped. Is this because it is difficult distinguish whether thread has not started in the first place or the thread has stopped.